docs/public/assets/bundle.js

```diff
@@ -73,20 +73,55 @@ module.exports = React.createClass({
 
   propTypes: {
     currencySymbol: React.PropTypes.string,
+    freeText: React.PropTypes.string,
+    removeText: React.PropTypes.string,
     price: React.PropTypes.number,
+    title: React.PropTypes.node,
+    toggleDescription: React.PropTypes.bool,
+    handleRemove: React.PropTypes.func,
     children: React.PropTypes.node
   },
+  getInitialState: function getInitialState() {
+    return {
+      descriptionVisibility: !this.props.toggleDescription
+    };
+  },
   getDefaultProps: function getDefaultProps() {
     return {
-      currencySymbol: '£'
+      title: null,
+      price: null,
+      handleRemove: null,
+      currencySymbol: '£',
+      freeText: 'FREE',
+      removeText: 'remove',
+      toggleDescription: false
     };
   },
-  render: function render() {
-    var priceNode;
-    if (this.props.price) {
-      priceNode = React.createElement('div', { className: 'component-basket-item-price' }, React.createElement('span', { className: 'component-currency' }, this.props.currencySymbol), this.props.price);
+  toggleDescriptionVisibility: function toggleDescriptionVisibility() {
+    this.setState({ descriptionVisibility: !this.state.descriptionVisibility });
+  },
+  titleNode: function titleNode() {
+    if (this.props.title === null) return null;
+    if (React.isValidElement(this.props.title)) return this.props.title;
+    if (this.props.toggleDescription) return React.createElement('a', { onClick: this.toggleDescriptionVisibility }, this.props.title);
+    return this.props.title;
+  },
+  priceNode: function priceNode() {
+    if (this.props.price === null) return null;
+    if (this.props.price === 0) {
+      return this.props.freeText;
     }
-    return React.createElement('div', { className: 'component-basket-item' }, React.createElement('div', { className: 'component-basket-item-description' }, this.props.children), priceNode);
+    return React.createElement('span', null, React.createElement('span', { className: 'component-basket-item-currency' }, this.props.currencySymbol), React.createElement('span', { className: 'component-basket-item-price' }, this.props.price));
```

Comment:
This is a generated file @pmcnr-hx - hopefully a future PR can remove the need for it. If not, [linguist](https://github.com/github/linguist#linguist) is GitHub's tool for, among other things, deciding which diffs need to be suppressed when PRs are displayed and it looks like we can create a dotfile to give it hints so this would not appear to require a review.